### PR TITLE
Time format problem solution

### PR DESCRIPTION
**Pull request by yuriihohan**

### CODE REVIEW

File: time_format/duration.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+require 'rubygems'
+require 'active_support/inflector'
+
+# Utilitity which provides human readable durations for seconds
+class Duration
```
Comment (vlebedeff) [inline time_format/duration.rb]: Maybe call it `HumanizedDuration` in order to communicate the intent with the class name itself.

File: time_format/duration.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  private def humanize
+    seconds_left = @seconds
+
+    seconds_left = humanize_part(seconds_left, IN_DAY, 'Day')
+    seconds_left = humanize_part(seconds_left, IN_HOUR, 'Hour')
+    seconds_left = humanize_part(seconds_left, IN_MINUTE, 'Minute')
+    humanize_part(seconds_left, IN_SECOND, 'Seconds', ' and')
+  end
+
+  private def humanize_part(seconds_left, interval_length, label, separator = ',')
```
Comment (vlebedeff) [inline time_format/duration.rb]: Despite the fact that putting `private` before every method is very explicit, it's more common to put one `private` modifier followed by all the private methods in the end of a class in Ruby.

Comment (vlebedeff) [inline time_format/duration.rb]: Naming nitpick: `def humanize_part(seconds, unit_length, unit_label, separator = ', ')` to emphasize the relation of the length and label in terms of unit.

File: time_format/duration.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    if seconds_left / interval_length > 0
+      units = seconds_left / interval_length
```
Comment (vlebedeff) [inline time_format/duration.rb]: It feels like unnecessary arithmetic is performed here
You could calculate the integral units and the remainder in one call with:
```
integral_units, remainder = seconds_left.divmod(interval_length)
```

File: time_format/duration.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      @humanized_string = concat_string @humanized_string, (pluralize_with_label units, label), separator
+      seconds_left -= (seconds_left / interval_length) * interval_length
+    end
+    seconds_left
+  end
+
+  private def pluralize_with_label(units, label)
+    "#{units} #{units > 1 ? label.pluralize : label}"
```
Comment (vlebedeff) [inline time_format/duration.rb]: If you're already using ActiveSupport library, you could just do `label.pluralize(units)`. It does comparison with `1`

File: time_format/duration.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    seconds_left = humanize_part(seconds_left, IN_DAY, 'Day')
+    seconds_left = humanize_part(seconds_left, IN_HOUR, 'Hour')
+    seconds_left = humanize_part(seconds_left, IN_MINUTE, 'Minute')
+    humanize_part(seconds_left, IN_SECOND, 'Seconds', ' and')
+  end
+
+  private def humanize_part(seconds_left, interval_length, label, separator = ',')
+    if seconds_left / interval_length > 0
+      units = seconds_left / interval_length
+      @humanized_string = concat_string @humanized_string, (pluralize_with_label units, label), separator
```
Comment (vlebedeff) [inline time_format/duration.rb]: It's better to use explicit braces in such cases:
```ruby
concat_string(@humanized_string, pluralize_with_label(units, label), separator)
```

weird that Rubocop didn't complain here

File: time_format/duration.rb (hunk 1 above)
Comment (yuriihohan) [reply time_format/duration.rb]: Agree

File: time_format/duration.rb (hunk 2 above)
Comment (yuriihohan) [reply time_format/duration.rb]: Done

File: time_format/duration.rb (hunk 3 above)
Comment (yuriihohan) [reply time_format/duration.rb]: Done

File: time_format/duration.rb (hunk 5 above)
Comment (yuriihohan) [reply time_format/duration.rb]: Fixed

File: time_format/duration.rb (hunk 4 above)
Comment (yuriihohan) [reply time_format/duration.rb]: fixed, much cleaner now

**Review by vlebedeff — APPROVED** [review]

**Pull request merged**